iotdb-core/datanode/src/main/java/org/apache/iotdb/db/protocol/client/ConfigNodeClient.java

```diff
@@ -1116,4 +1129,37 @@ public boolean validateObject(
           .orElse(false);
     }
   }
+
+  private class Cursor {
+    private int value;
+    private boolean locked = false;
+
+    public Cursor(int value) {
+      this.value = value;
+    }
+
+    public int get() {
+      return value;
+    }
+
+    public void set(int value) {
+      if (!locked) {
+        this.value = value;
+      } else {
+        logger.warn(
+            "ConfigNodeClient cursor will keep {} and won't change to {}, because it has been locked.",
+            configNodeEndPoints.get(this.value),
+            configNodeEndPoints.get(value));
+      }
+    }
+
+    public void setAndLock(int value) {
+      set(value);
+      locked = true;
+    }
+
+    public void unlock() {
+      locked = false;
+    }
```

Comment:
Is it necessary to change the value back when unlocking?

Reply:
Seems not necessary, because ConfigNodeClient can find leader by it self